/**
 * This file is part of the "sensord" project
 *   Copyright (c) 2015 Finn Zirngibl
 *   Copyright (c) 2015 Paul Asmuth
 *
 * FnordMetric is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License v3.0. You should have received a
 * copy of the GNU General Public License along with this program. If not, see
 * <http://www.gnu.org/licenses/>.
 */
package sensord;

message SampleEnvelope {

  /**
   * The optional namespace in which to which this sample should be reported
   */
  optional string namespace = 1;

  /**
   * The fully path to the used schema. e.g. "sensord.HostStats"
   */
  required string schema_name = 2;

  /**
   * The actual sample data, encoded with the schema named in schema_name
   */
  required string data = 3;

  /**
   * The time at which this sample was recorded as a unix microsecond timestamp
   */
  optional uint64 time = 4;

}

message SampleEnvelopeList {
  repeated SampleEnvelope smaples = 1;
}
